### PR TITLE
First Version

### PR DESCRIPTION
**Pull request by SamuelYvon**
My first submission. Using memory-mapping to split the file parsing in chunks, the rest is pretty standard stuff. Using manual parsing of lines to avoid building huge UTF-8 strings.

I have a few ideas where I can save a bit more time, especially for avoiding the String parsing / unparsing & Hashing. I think I can plug the vector API to parse the digits faster but I'm not sure if it's really going to be worth it.

#### Check List:
- [X] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [X] All formatting changes by the build are committed
- [X] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [X] Output matches that of `calculate_average_baseline.sh`
* Execution time: ~6s
* Execution time of reference implementation: ~2m 10

Note that I have a test that fails (with UTF-8 chars) but when I tested on other implementations it also failed for all of them, so I'm assuming it was an error in the provided code, I was a dozen commits behind. Comparing against the baseline yields correct results.

### CODE REVIEW

File: calculate_average_samuelyvon.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+
+source "$HOME/.sdkman/bin/sdkman-init.sh"
+sdk use java 21.0.1-graal 1>&2
+
+JAVA_OPTS="--enable-preview"
+time java $JAVA_OPTS --class-path target/average-1.0.0-SNAPSHOT.jar dev.morling.onebrc.CalculateAverage_samuelyvon
```
Comment (gunnarmorling) [inline calculate_average_samuelyvon.sh:23]:
```suggestion
java $JAVA_OPTS --class-path target/average-1.0.0-SNAPSHOT.jar dev.morling.onebrc.CalculateAverage_samuelyvon
```

Not needed any longer.

File: calculate_average_samuelyvon.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+
+source "$HOME/.sdkman/bin/sdkman-init.sh"
+sdk use java 21.0.1-graal 1>&2
```
Comment (gunnarmorling) [inline calculate_average_samuelyvon.sh:20]: Can you please move this to a new script named prepare_SamuelYvon.sh (also adjust the casing of this script to match your GH handle).

**Review by gunnarmorling — CHANGES_REQUESTED** [review]
This currently fails tests:

```
Validating calculate_average_samuelyvon.sh -- src/test/resources/samples/measurements-3.txt

Using java version 21.0.1-graal in this shell.

real	0m0.114s
user	0m0.211s
sys	0m0.050s
1c1
< {Bosaso=-15.0/1.2/20.0, Petropavlovsk-Kamchatsky=-9.5/0.0/9.5}
---
> {Bosaso=-15.0/1.3/20.0, Petropavlovsk-Kamchatsky=-9.5/0.0/9.5}
```

**Comment by SamuelYvon** [comment]
> This currently fails tests:
> 
> ```
> Validating calculate_average_samuelyvon.sh -- src/test/resources/samples/measurements-3.txt
> 
> Using java version 21.0.1-graal in this shell.
> 
> real	0m0.114s
> user	0m0.211s
> sys	0m0.050s
> 1c1
> < {Bosaso=-15.0/1.2/20.0, Petropavlovsk-Kamchatsky=-9.5/0.0/9.5}
> ---
> > {Bosaso=-15.0/1.3/20.0, Petropavlovsk-Kamchatsky=-9.5/0.0/9.5}
> ```

Little rounding error. This should be fixed. I have an error with an UTF-8 test but it seems non-sensical to me. Let me know if it happens on your test setup as well (it fails for other submissions as well locally)

File: calculate_average_samuelyvon.sh (hunk 2 above)
Comment (SamuelYvon) [reply calculate_average_samuelyvon.sh]: Missed the casing for the username... hold on... I was sure it was all lowercased >_<

**Comment by SamuelYvon** [comment]
@gunnarmorling Ok, applied your changes.

I still have errors running this test: `src/test/resources/samples/measurements-20.txt`.
Unsure why the output expects the UTF-8 bee when it does not appear in the input. Odd :( It fails for other submissions so I'm unsure what's going on (same on "royvanrijn"'s submission)

**Comment by gunnarmorling** [comment]
All looking good now, tests pass too. 00:12.471, nice! Thx for participating!

**Pull request merged**